packages/flutter_tools/lib/src/android/android_device.dart

```diff
@@ -354,36 +354,40 @@ class AndroidDevice extends Device {
       // device has printed "Observatory is listening on...".
       printTrace('Waiting for observatory port to be available...');
 
+      // TODO(danrubel): The iOS device class does something similar to this code below.
+      // The various Device subclasses should be refactored and common code moved into the superclass.
       try {
-        int observatoryDevicePort, diagnosticDevicePort;
+        Uri observatoryDeviceUri, diagnosticDeviceUri;
 
         if (debuggingOptions.buildMode == BuildMode.debug) {
-          Future<List<int>> scrapeServicePorts = Future.wait(
-              <Future<int>>[observatoryDiscovery.nextPort(), diagnosticDiscovery.nextPort()]
+          Future<List<Uri>> scrapeServiceUris = Future.wait(
+              <Future<Uri>>[observatoryDiscovery.nextUri(), diagnosticDiscovery.nextUri()]
           );
-          List<int> devicePorts = await scrapeServicePorts.timeout(new Duration(seconds: 20));
-          observatoryDevicePort = devicePorts[0];
-          diagnosticDevicePort = devicePorts[1];
+          List<Uri> deviceUris = await scrapeServiceUris.timeout(new Duration(seconds: 20));
+          observatoryDeviceUri = deviceUris[0];
+          diagnosticDeviceUri = deviceUris[1];
         } else {
```

Comment:
I think this should be else if (debuggingOptions.buildMode == BuildMode.profile)

in release mode you will never get a print.